pkg/name/name.go

```diff
@@ -67,87 +67,78 @@ const (
 // ManifestMap is a map of ComponentName to its manifest string.
 type ManifestMap map[ComponentName]string
 
-// IsComponentEnabled reports whether the given feature and component are enabled in the given spec. The logic is, in
+// IsComponentEnabledInSpec reports whether the given feature and component are enabled in the given spec. The logic is, in
 // order of evaluation:
 // 1. if the feature is not defined, the component is disabled, else
 // 2. if the feature is disabled, the component is disabled, else
 // 3. if the component is not defined, it is reported disabled, else
 // 4. if the component disabled, it is reported disabled, else
 // 5. the component is enabled.
 // This follows the logic description in IstioControlPlane proto.
-func IsComponentEnabled(featureName FeatureName, componentName ComponentName, installSpec *v1alpha2.IstioControlPlaneSpec) bool {
+func IsComponentEnabledInSpec(featureName FeatureName, componentName ComponentName, installSpec *v1alpha2.IstioControlPlaneSpec) (bool, error) {
```

Comment:
`IsComponentEnabledInSpec` and `Namespace` functions mix user input validation and runtime behavior. Long term it would be good to separate these into distinct phases so we can catch bad input earlier in the pipeline.

Reply:
Agreed and I had exactly the same feeling as I was working on this. I do feel uneasy about not double checking, but if the contract is clear the extra check can be removed. I've put a TODO here for now because validation is far from perfect right now so more work is needed before we can be sure the inputs are good.